### DISCUSSION TITLE
Timelines, characters and themes disappeared in dialogic 1.5

### DISCUSSION THREAD

**Discussion by Chrishoper** [dialogic-godot/dialogic#1876] [Q&A] [answered]
Hello:
I am editing my timeline, character, and theme using dialogic 1.5 of godot 3.5.3, and I can see the json files in the dialogic folder on the local path, but they are missing in dialogic. I was able to create a new timeline using dialogic as normal, and I was able to find its corresponding json file in the local folder. But why can't I see the timeline I originally edited? I didn't use the latest version of dialogic because I couldn't use it to customize my theme well.

Thank you very much for your help!

system：
OS: win10
Godot Version: 3.5.3
Dialogic Version: 1.5

**Comment by Iodinex64** [comment, marked as answer]
I had a similar issue recently where I duplicated a theme, but that theme was not shown in the Dialogic GUI. Here's what I think is happening.

In Dialogic 1.5, there's a JSON file called "folder_structure.json". This file tells Dialogic about every timeline, character, theme etc. that exists, and how to display them. If we have a JSON file for a character or theme and it is NOT present in this JSON file, Dialogic will not display it in Godot.

When we operate on characters, themes or whatever else in Dialogic, this folder_structure.json file gets updated. It must be the case that somewhere in Dialogic, it's not being updated properly when we operate in certain ways (eg. duplicate).

A workaround right now would be to open folder_structure.json (located in the root of the "dialogic" folder) in a text editor, find the Characters or Themes heading, and add the file names of the missing character/theme .json's/.cfg's yourself. They should then show up in the editor.

**Reply by Chrishoper** [reply]
Thanks for the suggestion!
I tried to add the json name corresponding to my timeline to the folder_structure json file according to the method you said, but when I opened the editor, I still couldn't see the corresponding timeline and other components (totally blank), and when I created a new timeline in the editor, the local folder corresponds to the folder_ structure. json will be refreshed, and only the new timeline will be displayed.
Did you edit the folder_structure. json with the editor opened or not?  Did it work?
Again, your help is much appreciated!!☆⌒(*＾-゜)v THX!!

**Reply by Iodinex64** [reply]
Hi there.

I closed the editor completely, then added the name of the timeline to folder_structure.json where it should be. I then opened the editor again.

In 1.5, I believe folder_structure.json gets written to quite a bit, so it's possible that if the engine was open while you were editing the json, the engine could have over-written your changes.

Doing it while Godot is closed should help. Let me know how that goes.

**Reply by Chrishoper** [reply]
I did it! 

The problem should be that I only edited the timeline into the folder_structure json file, the theme and character files that the timeline relies on were not edited correctly. If someone else encounters similar issues, you can edit all local files into the folder_structure json file to avoid missing necessary components.

Once again, thank you very much for your help. You are a great help.
